cmake/configure_hera.gnu.cmake

```diff
@@ -1,37 +1,11 @@
-message("")
-message("Setting configuration for $ENV{CMAKE_Platform}")
-message("")
-
-get_filename_component (C_COMPILER_NAME ${CMAKE_C_COMPILER} NAME)
-get_filename_component (CXX_COMPILER_NAME ${CMAKE_CXX_COMPILER} NAME)
-get_filename_component (Fortran_COMPILER_NAME ${CMAKE_Fortran_COMPILER} NAME)
-message("C       compiler: ${CMAKE_C_COMPILER_ID} ${CMAKE_C_COMPILER_VERSION} (${C_COMPILER_NAME})")
-message("CXX     compiler: ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION} (${CXX_COMPILER_NAME})")
-message("Fortran compiler: ${CMAKE_Fortran_COMPILER_ID} ${CMAKE_Fortran_COMPILER_VERSION} (${Fortran_COMPILER_NAME})")
-message("")
-
-option(DEBUG   "Enable DEBUG mode" OFF)
-option(REPRO   "Enable REPRO mode" OFF)
-option(VERBOSE "Enable VERBOSE mode" OFF)
-option(32BIT   "Enable 32BIT (single precision arithmetic in dycore)" OFF)
-option(OPENMP  "Enable OpenMP threading" ON)
-option(AVX2    "Enable AVX2 instruction set" OFF)
+option(32BIT           "Enable 32BIT (single precision arithmetic in dycore)" OFF)
```

Comment:
rather than have the same information in all the `configure_${PLATFORM}.${compiler}.cmake`,
can we reduce this to 2 sets?
`configure.gnu.cmake` and `configure.intel.cmake`.
If there is a platform that has something that needs something different that the "stock" options, only those options can be placed in a file with `${PLATFORM}`

Reply:
We (@climbfuji and @junwang-noaa) discussed this and decided that to keep all options defined in separate files for all 'Tier-1' platforms. Your suggestion tho have `configure.gnu.cmake` and `configure.intel.cmake` is basically what is now `configure_linux.gnu.cmake` and `configure_linux.intel.cmake` (ie. generic Linux platforms)

Reply:
and why that is not sufficient? Are Tier-1 platforms not built on generic Linux platforms, give or take some options e.g. AVX2 or inline POST

Reply:
When we discussed this initially, the suggestion was to only have the tier-1 platform configuration and users would have to always specify deviations from it on the command line. This is not ok, the model should build out of the box on other systems as well. Rahul's suggestion will work, and so does Dusan's. As long as users do not have to manually specify `-DINLINE_POST=OFF` every time the build the model, I don't care which version we use.

Reply:
My original suggestion was to just put all options in the top-level `CMakeLists.txt`. Because `configure_linux.gnu.cmake` and `configure_linux.intel.cmake` are identical we do not need them both.

Reply:
@DusanJovic-NOAA 
I second that suggestion.
Looking ahead with a coupled model, these options are just going to increase and become too many. And placing/repeating them in all platform/tier specific files is going to get messy, hard to maintain and prone to errors.

Reply:
How about this. In the top-level `CMakeLists.txt` we define these options (with the default values equal to what's currently in 'generic' linux configuration files):
```
set(32BIT           OFF CACHE BOOL "Enable 32BIT (single precision arithmetic in dycore)")
set(AVX2            ON  CACHE BOOL "Enable AVX2 instruction set")
set(CCPP            ON  CACHE BOOL "Enable CCPP")
set(DEBUG           OFF CACHE BOOL "Enable DEBUG mode")
set(INLINE_POST     OFF CACHE BOOL "Enable inline post")
set(MULTI_GASES     OFF CACHE BOOL "Enable MULTI_GASES")
set(OPENMP          ON  CACHE BOOL "Enable OpenMP threading")
set(PARALLEL_NETCDF OFF CACHE BOOL "Enable parallel NetCDF")
set(QUAD_PRECISION  ON  CACHE BOOL "Enable QUAD_PRECISION (for certain grid metric terms in dycore)")
set(REPRO           OFF CACHE BOOL "Enable REPRO mode")
set(WW3             OFF CACHE BOOL "Enable WW3")
```
Then based on `CMAKE_Platform`, if file exists, we include platform specific file:
```
if(EXISTS ${PROJECT_SOURCE_DIR}/cmake/configure_${CMAKE_Platform}.cmake)
  message("")
  message("Setting configuration for $ENV{CMAKE_Platform}")
  message("")
  include(${PROJECT_SOURCE_DIR}/cmake/configure_${CMAKE_Platform}.cmake)
endif()
```
On Hera, for example, this file will be:
```
$ cat cmake/configure_hera.intel.cmake 
set(INLINE_POST     ON CACHE BOOL "Enable inline post" FORCE)
set(PARALLEL_NETCDF ON CACHE BOOL "Enable parallel NetCDF" FORCE)
```
which will override `INLINE_POST` and `PARALLEL_NETCDF`.

Reply:
That's fine with me. How about compiler-dependent options? AVX2 only makes sense if Intel is used, for GNU that option doesn't exist (or could be replaced by another option).

Reply:
We do not use AVX2 option to define architecture specific compiler flags for GNU, but GNU compilers do have such options (`-march=...`). Maybe we'll start using them.
https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html